R/estimate_KM.R

```diff
@@ -65,102 +65,56 @@
 #' visR::estimate_KM(data = veteran_adam, strata = "trt")
 
 estimate_KM <- function(
-   data = NULL
-  ,strata = NULL
-  ,CNSR = "CNSR"
-  ,AVAL = "AVAL"
-  ,...
+    data = NULL
+    ,strata = NULL
+    ,CNSR = "CNSR"
+    ,AVAL = "AVAL"
+    ,...
 ){
-
-# Capture input to validate user input for data argument -----------------
+  # Capture input to validate user input for data argument ---------------------
   dots <- rlang::dots_list(...)
 
- ## Get actual data name as symbol
- ### Magrittre pipe returns "." which inactivates recalls to survfit in downstream functions
- ### map passes .x as Call$data
- ### df: catch expressions that represent base R subsets
-  Call <- as.list(match.call())
-  dfExpr <- Call[["data"]]
-
- ## Validate `data` and capture data name
-
-  if (is.null(data)) stop(paste0("Data can't be NULL."))
-
-  if (base::length(base::deparse(Call[["data"]])) == 1 && base::deparse(Call[["data"]]) %in% c(".", ".x")){
-    df <- the_lhs()
-    Call[["data"]] <- as.symbol(df)
-  } else {
-     df <- as.character(sub("\\[.*$", "", deparse(dfExpr))[1])
-  }
-
-  if (!(inherits(data, "data.frame") | inherits(data, "tibble") | inherits(data, "data.table"))) stop(paste0("Data can be of `class` dataframe or tibble or data.table."))
-
-  data <- as.data.frame(data)
-
-# Validate columns --------------------------------------------------------
+  ## Validate argument inputs --------------------------------------------------
+  if (is.null(data))
+    stop(paste0("Data can't be NULL."))
+  if (!is.data.frame(data))
+    stop("Data can be of `class` dataframe or tibble or data.table.")
 
   reqcols <- c(strata, CNSR, AVAL)
-
   if (! all(reqcols %in% colnames(data))){
     stop(paste0("Following columns are missing from `data`: ", paste(setdiff(reqcols, colnames(data)), collapse = " "), "."))
   }
 
-  if (! is.numeric(data[[AVAL]])){
+  if (!is.numeric(data[[AVAL]])){
     stop("Analysis variable (AVAL) is not numeric.")
   }
 
-  if (! is.numeric(data[[CNSR]])){
+  if (!is.numeric(data[[CNSR]])){
     stop("Censor variable (CNSR) is not numeric.")
   }
 
-# Ensure the presence of at least one strata -----------------------------
```

Comment:
I prefer this explicit code section for easy debugging and understanding of what happens rather than to be more compact. Our target audience included SAS programmers that have little experience in R

Reply:
If you prefer formula_rhs instead of main, that is OK for me

Reply:
Please feel free to change as you see fit. I will note that `tidyr::drop_na(any_of(c(AVAL, CNSR, strata)))` only uses top-level, user-facing functions: no behind-the-scenes-type functions here